packages/components/src/responsive-toolbar/index.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, cloneElement, Children, createRef } from '@wordpress/element';
+import { withInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import IconButton from '../icon-button';
+import Dropdown from '../dropdown';
+import Disabled from '../disabled';
+
+/**
+ * Module constants
+ */
+const OFFSET = 60;
+
+class ResponsiveToolbar extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			countHiddenChildren: 0,
+		};
+		this.container = createRef();
+		this.hiddenContainer = createRef();
+
+		this.throttledUpdateHiddenItems = this.throttledUpdateHiddenItems.bind( this );
+	}
+
+	componentDidMount() {
+		this.toggleWindowEvents( true );
+		this.updateHiddenItems();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.children !== this.props.children ) {
+			this.updateHiddenItems();
+		}
+	}
+
+	componentWillUnmount() {
+		this.toggleWindowEvents( false );
+	}
+
+	toggleWindowEvents( isListening ) {
+		const handler = isListening ? 'addEventListener' : 'removeEventListener';
+
+		window.cancelAnimationFrame( this.rafHandle );
+		window[ handler ]( 'resize', this.throttledUpdateHiddenItems );
+		window[ handler ]( 'scroll', this.throttledUpdateHiddenItems, true );
+	}
+
+	throttledUpdateHiddenItems() {
+		this.rafHandle = window.requestAnimationFrame( () => this.updateHiddenItems() );
+	}
+
+	updateHiddenItems() {
+		const { instanceId } = this.props;
+		const containerRect = this.container.current.getBoundingClientRect();
+		let countHiddenChildren = 0;
+		const total = this.hiddenContainer.current.childNodes.length;
+		this.hiddenContainer.current.childNodes.forEach( ( child ) => {
+			const childRect = child.getBoundingClientRect();
+			if (
+				childRect.left < containerRect.left ||
+				childRect.right > containerRect.right - OFFSET
+			) {
+				countHiddenChildren++;
```

Comment:
Minor: This would make sense as a `reduce`, though I suppose `childNodes` is a `NodeList`, not `Array`, so not natively available without either using Lodash or converting to array `[ ...childNodes ]`:

```js
const total = childNodes.reduce( ( result, child ) => {
	// ...
}, 0 );
```